l2/system-contracts/TransactionHelper.sol

```diff
@@ -33,15 +46,25 @@ struct Transaction {
 	// it would allow easier proof integration (in case we will need
 	// some special circuit for preprocessing transactions).
 	uint256[6] reserved;
```

Comment:
why not include nonce and value as proper fields? makes it easier to understand in code than handles `Transaction` structs

Reply:
Great idea! Firstly we introduced the transaction format and some time after added the native `msg.value`. So, it is kinda legacy code

Reply:
i'm ok with doing breaking changes like those and making it clearer